CDTDatastore/HTTP/CDTSessionCookieInterceptorBase.m

```diff
@@ -142,12 +145,24 @@ Handles retrieving a cookie ("logging in") for the credentials this interceptor
                                           self.shouldMakeSessionRequest = NO;
                                       } else {
                                           // Most other HTTP status codes are non-transient failures; don't retry.
-                                          CDTLogError(CDTREPLICATION_LOG_CONTEXT,
+                                          NSString *dataJsonResponse = [[NSString alloc] initWithData:data
+                                                                                             encoding:NSUTF8StringEncoding];
+                                          if(dataJsonResponse) {
+                                            CDTLogError(CDTREPLICATION_LOG_CONTEXT,
```

Comment:
I guess inside here still needs a check for status code 403 and "credentials_expired" in the error response. This will also affect `self.shouldMakeSessionRequest` and the `authentication will not be attempted again by this interceptor` message

Reply:
@tomblench Would it be better for 403 to have it's own if block just like the 401 case?
```
...
else if (httpResp.statusCode == 403) {
  // Credentials expired; retry.
  NSString *dataJsonResponse = [[NSString alloc] initWithData:data
                                                     encoding:NSUTF8StringEncoding];
  if(!dataJsonResponse) {
      dataJsonResponse = @"{\"error\": \"credentials_expired\", \"reason\": \"Session expired\"}";
  }
  
  CDTLogError(CDTREPLICATION_LOG_CONTEXT, @"Credentials expired, response: %@", dataJsonResponse);
  self.shouldMakeSessionRequest = NO;
}
```

Reply:
Yes, makes sense

Reply:
It's worth noting that `403` with `credentials_expired` only applies for `_session` not `_iam_session`. So the `403` block might be better placed in `CDTSessionCookieInterceptor` than in the Base class. I don't think it hurts too much to have it in common, but it will mean doing unnecessary checks on `403` responses for `_iam_session`.

Reply:
Updated in f8fb921.

Reply:
@tomblench @ricellis Removed this code after Slack conversation and updated the 403 if block case in `interceptResponseInContext` in ebe1749.